test/Npgsql.Tests/SystemTransactionTests.cs

```diff
@@ -341,7 +489,7 @@ public void SchemaConnectionShouldntEnlist()
         }
 
         void AssertNoPreparedTransactions()
-            => Assert.That(GetNumberOfPreparedTransactions(), Is.EqualTo(0));
+            => Assert.That(GetNumberOfPreparedTransactions(), Is.EqualTo(0), "Prepared transactions found");
 
         int GetNumberOfPreparedTransactions()
```

Comment:
Do you mean that you were seeing 0 transactions in pg_prepared_xacts although you were expecting one to be there? Do you consider this as evidence of the race condition you were mentioning before (or another race condition)?